introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "\n",
+    "objective_metric_name = 'test:RMSE'\n",
+    "\n",
+    "tuner = HyperparameterTuner(deepar,\n",
+    "                            objective_metric_name,\n",
+    "                            hyperparameter_ranges,\n",
+    "                            max_jobs=10,\n",
+    "                            strategy='Bayesian',\n",
+    "                            objective_type='Minimize',\n",
+    "                            max_parallel_jobs=10,\n",
+    "                            early_stopping_type='Auto')\n",
+    "\n",
+    "s3_input_train = sagemaker.s3_input(s3_data='s3://{}/{}/train/'.format(bucket, prefix),\n",
+    "                                    content_type='json')\n",
+    "s3_input_test = sagemaker.s3_input(s3_data='s3://{}/{}/test/'.format(bucket, prefix),\n",
+    "                                   content_type='json')\n",
+    "\n",
+    "tuner.fit({'train': s3_input_train, 'test': s3_input_test}, \n",
+    "          include_cls_metadata=False)\n",
+    "job_name = tuner.latest_tuning_job.name\n",
+    "tuner.wait()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## Host\n",
+    "\n",
+    "We use the [HyperParameterTuner](https://sagemaker.readthedocs.io/en/stable/tuner.html) to host the best model using a single ml.m4.xlarge instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "best_tuning_job_name = tuner.best_training_job()\n",
+    "endpoint_name = tuner.deploy(initial_instance_count=1,\n",
+    "                             endpoint_name=best_tuning_job_name,\n",
+    "                             instance_type='ml.m4.xlarge',\n",
+    "                             wait=True)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluate\n",
+    "\n",
+    "To evaluate the model, define a DeepARPredictor class. This class extends the [RealTimePredictor](https://sagemaker.readthedocs.io/en/stable/predictors.html) class. Implementing encode and decode function helps making requests using `pandas.Series` objects rather than raw JSON strings.\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class DeepARPredictor(sagemaker.predictor.RealTimePredictor):\n",
+    "    \n",
+    "    def set_prediction_parameters(self, freq, prediction_length):\n",
+    "        \"\"\"Set the time frequency and prediction length parameters. This method **must** be \n",
+    "        called before being able to use `predict`.\n",
+    "        \n",
+    "        Parameters:\n",
+    "        freq -- string indicating the time frequency\n",
+    "        prediction_length -- integer, number of predicted time points\n",
+    "       \n",
+    "       Return value: none.\n",
+    "       \"\"\"\n",
```

Comment:
these lines should be indented by one more space to align with the other lines

Reply:
fixed